src/python/pants/backend/python/subsystems/twine.py

```diff
@@ -26,14 +26,15 @@ class TwineSubsystem(PythonToolBase):
     name = "Twine"
     help = "The utility for publishing Python distributions to PyPI and other Python repositories."
 
-    default_version = "twine>=3.7.1,<3.8"
+    default_version = "twine>=4,<5"
     default_main = ConsoleScript("twine")
 
     # This explicit dependency resolves a weird behavior in poetry, where it would include a sys
     # platform constraint on "Windows" when this was included transitively from the twine
     # requirements.
     # See: https://github.com/pantsbuild/pants/pull/13594#issuecomment-968154931
     default_extra_requirements = ["colorama>=0.4.3"]
+    default_requirements = ["twine>=3.7.1,<5", *default_extra_requirements]
```

Comment:
`default_version != default_requirements[0]`

Reply:
Yes, by design, see my comment.

Reply:
If we wanted this to be the case uniformly we could just do it in the base class...